File: backend/Models/Users/teams/team.py
# coding=utf-8

from datetime import datetime
from sqlalchemy import Column, String, Integer, Float, DateTime
from sqlalchemy.dialects.postgresql import ARRAY
from ...sqlal_mutable_array import MutableList
from ...entity import Entity, DB_SESSION, ENGINE, BASE
from marshmallow import Schema, fields

class TeamTable(Entity, BASE):
    __tablename__ = 'teams'

    id = Column(Integer, primary_key=True)
    leader = Column(String)
    members = Column(MutableList.as_mutable(ARRAY(String)))
    name = Column(String, unique=True)
    challenges = Column(MutableList.as_mutable(ARRAY(Integer)))

    def __init__(self, leader, members, name, created_by):
        Entity.__init__(self, created_by)
        self.leader = leader
        self.members = members
        self.name = name

class TeamTableSchema(Schema):
    id = fields.Number()
    leader = fields.Str()
    members = fields.List(fields.Str())
    name = fields.Str()
    challenges = fields.List(fields.Number())

    created_at = fields.DateTime()
    updated_at = fields.DateTime()
    last_updated_by = fields.Str()

File: frontend/src/Helpers/Team/Classes/Team.ts
import { ITeam } from '../Interfaces/ITeam';

export class Team implements ITeam {
    private id: number;
    private name: string;
    private leader: string;
    private members: string[];
    private challenges: string[];
    private updatedAt: Date;
    private createdAt: Date;
    private lastUpdatedBy: string;

    constructor(name, leader, members) {
        this.name = name;
        this.leader = leader;
        this.members = members;
    }

    public get Id(): number {
        return this.id;
    }
    public set Id(value: number) {
        this.id = value;
    }

    public get Name(): string {
        return this.name;
    }
    public set Name(value: string) {
        this.name = value;
    }

    public get Leader(): string {
        return this.leader;
    }
    public set Leader(value: string) {
        this.leader = value;
    }


    public get Members(): string[] {
        return this.members;
    }
    public set Members(value: string[]) {
        this.members = value;
    }

    public get Challenges(): string[] {
        return this.challenges;
    }
    public set Challenges(value: string[]) {
        this.challenges = value;
    }

    public get UpdatedAt(): Date {
        return this.updatedAt;
    }
    public set UpdatedAt (value: Date) {
        this.updatedAt = value;
    }

    public get CreatedAt(): Date {
    return this.createdAt;
    }
    public set CreatedAt (value: Date) {
        this.createdAt = value;
    }

    public get LastUpdatedBy(): string {
    return this.lastUpdatedBy;
    }
    public set LastUpdatedBy (value: string) {
        this.lastUpdatedBy = value;
    }
}